src/pipecat/services/asyncai/tts.py

```diff
@@ -0,0 +1,513 @@
+            self._websocket = None
+            await self._call_event_handler("on_connection_error", f"{e}")
+
+    async def _disconnect_websocket(self):
+        try:
+            await self.stop_all_metrics()
+
+            if self._websocket:
+                logger.debug("Disconnecting from Async")
+                await self._websocket.close()
+        except Exception as e:
+            logger.error(f"{self} error closing websocket: {e}")
+        finally:
+            self._context_id = None
+            self._websocket = None
+
+    def _get_websocket(self):
+        if self._websocket:
+            return self._websocket
+        raise Exception("Websocket not connected")
+
+    async def _handle_interruption(self, frame: StartInterruptionFrame, direction: FrameDirection):
+        await super()._handle_interruption(frame, direction)
+        await self.stop_all_metrics()
+        if self._context_id:
+            self._context_id = None
+
+    async def flush_audio(self):
+        """Flush any pending audio and finalize the current context."""
+        if not self._context_id or not self._websocket:
+            return
+        logger.trace(f"{self}: flushing audio")
+        msg = self._build_msg(text=" ", force=True)
+        await self._websocket.send(msg)
+        self._context_id = None
+
+    async def _receive_messages(self):
+        async for message in WatchdogAsyncIterator(
+            self._get_websocket(), manager=self.task_manager
+        ):
+            msg = json.loads(message)
+            context_id = self._global_context_id
+            if not msg:
+                continue
+
+            if "final" in msg and msg["final"] is True:
+                await self.stop_ttfb_metrics()
+                await self.remove_audio_context(context_id)
+            elif msg.get("audio"):
+                await self.stop_ttfb_metrics()
+                frame = TTSAudioRawFrame(
+                    audio=base64.b64decode(msg["audio"]),
+                    sample_rate=self.sample_rate,
+                    num_channels=1,
+                )
+                await self.append_to_audio_context(context_id, frame)
+            elif msg.get("error_code"):
+                logger.error(f"{self} error: {msg}")
+                await self.push_frame(TTSStoppedFrame())
+                await self.stop_all_metrics()
+                await self.push_error(ErrorFrame(f"{self} error: {msg['message']}"))
+                self._context_id = None
+            else:
+                logger.error(f"{self} error, unknown message type: {msg}")
+
+    async def _keepalive_task_handler(self):
+        """Send periodic keepalive messages to maintain WebSocket connection."""
+        KEEPALIVE_SLEEP = 10 if self.task_manager.task_watchdog_enabled else 3
+        while True:
+            self.reset_watchdog()
+            await asyncio.sleep(KEEPALIVE_SLEEP)
+            try:
+                if self._websocket and self._websocket.open:
```

Comment:
Update for the asyncio implementation:

```suggestion
                if self._websocket and self._websocket.state is State.OPEN:
```